types/tailwindcss/tailwind-config.d.ts

```diff
@@ -0,0 +1,1188 @@
+export type Variant =
+    | 'responsive'
+    | 'first'
+    | 'last'
+    | 'odd'
+    | 'even'
+    | 'visited'
+    | 'checked'
+    | 'group-hover'
+    | 'group-focus'
+    | 'focus-within'
+    | 'hover'
+    | 'focus'
+    | 'focus-visible'
+    | 'active'
+    | 'disabled'
+    | 'dark';
+
+export interface TailwindColorGroup {
+    readonly '50': string;
+    readonly '100': string;
+    readonly '200': string;
+    readonly '300': string;
+    readonly '400': string;
+    readonly '500': string;
+    readonly '600': string;
+    readonly '700': string;
+    readonly '800': string;
+    readonly '900': string;
+}
+
+export interface TailwindStandardSpacing {
+    readonly '0': string;
+    readonly '1': string;
+    readonly '2': string;
+    readonly '3': string;
+    readonly '4': string;
+    readonly '5': string;
+    readonly '6': string;
+    readonly '7': string;
+    readonly '8': string;
+    readonly '9': string;
+    readonly '10': string;
+    readonly '11': string;
+    readonly '12': string;
+    readonly '14': string;
+    readonly '16': string;
+    readonly '20': string;
+    readonly '24': string;
+    readonly '28': string;
+    readonly '32': string;
+    readonly '36': string;
+    readonly '40': string;
+    readonly '44': string;
+    readonly '48': string;
+    readonly '52': string;
+    readonly '56': string;
+    readonly '60': string;
+    readonly '64': string;
+    readonly '72': string;
+    readonly '80': string;
+    readonly '96': string;
+    readonly px: string;
+    readonly '0.5': string;
+    readonly '1.5': string;
+    readonly '2.5': string;
+    readonly '3.5': string;
+}
+
+export type TailwindStandardNegativeSpacing = {
+    [key in keyof TailwindStandardSpacing as `-${key}`]: string;
+};
+
+export interface TailwindAnimationConfig {
+    [key: string]: {
+        transform?: string;
+        opacity?: string;
+        animationTimingFunction?: string;
+    };
+}
+
+export type TailwindFontConfig = [
+    string,
+    {
+        lineHeight: string;
+    },
+];
+
+export interface TailwindColorConfig {
+    readonly transparent: string;
+    readonly current: string;
+    readonly black: string;
+    readonly white: string;
+    readonly gray: TailwindColorGroup;
+    readonly red: TailwindColorGroup;
+    readonly yellow: TailwindColorGroup;
+    readonly green: TailwindColorGroup;
+    readonly blue: TailwindColorGroup;
+    readonly indigo: TailwindColorGroup;
+    readonly purple: TailwindColorGroup;
+    readonly pink: TailwindColorGroup;
+}
+
+export interface TailwindConfig {
+    readonly theme: {
+        readonly screens: {
+            readonly sm: string;
+            readonly md: string;
+            readonly lg: string;
+            readonly xl: string;
+            readonly '2xl': string;
+            readonly [key: string]: string;
+        };
+        readonly colors: TailwindColorConfig;
```

Comment:
This typing seems to capture what tailwind config returns by default, but it doesn't allow customizing the tailwind config if I understand correctly. Is that the purpose of these types?  What about customised tailwind configurations? Custom color names are allowed by Tailwind implementation but not in these types.

In our use case, it would be useful to have the the structure of the config typed, but not limiting the exact names of e.g. colors, breakpoints etc that are customisable.

Reply:
Correct. Tailwind allows for extentions

I believe adding `readonly [key: string]: string;` will allow for extending you speak of:

> "but not limiting the exact names"

https://tailwindcss.com/docs/customizing-colors#extending-the-defaults

Reply:
@dolanmiu If I'm reading your response correctly, it sounds like you were planning to add a new member, but I don't see a new commit.  Have I misunderstood?

Reply:
What do you mean by "new member"?
The members can be whatever you want due to the `readonly [key: string]: string;` syntax, you could call it `blah`, and its valid, and can be used in Tailwind if you wish.

Reply:
@dolanmiu From your comment, it sounded like you were going to add `readonly [key: string]: string;` to `TailwindColorConfig` (or use an intersection, as for spacing, below).  If you are not planning to do so, can you please elaborate on your explanation of how custom colors will be allowed (or, better still, add a unit test demonstrating that it is possible)?